keras_cv/src/models/object_detection/mask_rcnn/faster_rcnn_backbone.py

```diff
@@ -0,0 +1,812 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tree
+
+from keras_cv.src import losses
+#from keras_cv.src.api_export import keras_cv_export
+from keras_cv.src.backend import keras
+from keras_cv.src.backend import ops
+from keras_cv.src.bounding_box import convert_format
+from keras_cv.src.bounding_box.converters import _decode_deltas_to_boxes
+from keras_cv.src.bounding_box.utils import _clip_boxes
+from keras_cv.src.layers.object_detection.anchor_generator import (
+    AnchorGenerator,
+)
+from keras_cv.src.layers.object_detection.box_matcher import BoxMatcher
+from keras_cv.src.layers.object_detection.roi_align import ROIAligner
+from keras_cv.src.layers.object_detection.roi_generator import ROIGenerator
+from keras_cv.src.layers.object_detection.rpn_label_encoder import (
+    RpnLabelEncoder,
+)
+from keras_cv.src.models.object_detection.faster_rcnn import FeaturePyramid
+from keras_cv.src.models.object_detection.faster_rcnn import RCNNHead
+from keras_cv.src.models.object_detection.faster_rcnn import RPNHead
+from keras_cv.src.models.object_detection.mask_rcnn.non_max_suppression import (
+    NonMaxSuppression,
+)
+from keras_cv.src.models.object_detection.mask_rcnn.roi_sampler import (
+    ROISampler,
+)
+from keras_cv.src.models.task import Task
+from keras_cv.src.utils.train import get_feature_extractor
+
+BOX_VARIANCE = [0.1, 0.1, 0.2, 0.2]
+
+
+# @keras_cv_export(
+#     [
+#         "keras_cv.models.FasterRCNN",
+#         "keras_cv.models.object_detection.FasterRCNN",
+#     ]
+# )
+class FasterRCNN(Task):
```

Comment:
This should subclass backbone class. And the whole model implementation need not be added here. This could just be a simple wrapper for the FasterRCNN backbone